Sources/MongoSwift/MongoClient.swift

```diff
@@ -14,6 +14,10 @@ public struct MongoClientOptions: CodingStrategyProvider {
     /// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/#urioption.compressors
     public var compressors: [Compressor]?
 
+    /// Specifies the time, in milliseconds, for an individual connection from your connection pool to establish a TCP
+    /// connection to the MongoDB server before timing out.
+    public var connectTimeoutMS: ChannelOptions.Types.ConnectTimeoutOption?
```

Comment:
I like the idea of using a NIO type for this, but I think for consistency with our other timeout options we'll need to continue to use just `Int` here. In 2.0 (the pure swift version), I think we'd probably want to update these various timeouts to use such a type, though I think we might opt for the more general `TimeAmount` instead, which `ChannelOptions.Types.ConnectTimeoutOption` is basically an alias for. Could you file a ticket to remind us to do this in 2.0?

BTW, the reason we have to wait for 2.0 to update the types is that the driver follows [Semantic Versioning](https://semver.org/), and since changing the type of existing options would constitute a breaking change, we'd only be able to introduce it in a new major version. We're currently not planning on doing another major version release until the new driver is done, though.

Reply:
done